mlflow/models/evaluation/default_evaluator.py

```diff
@@ -41,20 +42,30 @@
 _DEFAULT_SAMPLE_ROWS_FOR_SHAP = 2000
 
 
-def _infer_model_type_by_labels(labels):
-    distinct_labels = set(labels)
-    for v in distinct_labels:
+def _is_categorical(values):
+    distinct_values = set(values)
+    for v in distinct_values:
         if not isinstance(v, numbers.Number):
-            return "classifier"
+            return True
         if not float(v).is_integer():
-            return "regressor"
-    if len(distinct_labels) > 1000 and len(distinct_labels) / len(labels) > 0.7:
-        return "regressor"
-    if len(distinct_labels) <= 20:
-        return "classifier"
+            return False
+    if len(distinct_values) > 1000 and len(distinct_values) / len(values) > 0.7:
+        return False
+    if len(distinct_values) <= 20:
+        return True
     return None  # unknown
```

Comment:
What does None mean here? Can we just return either True or False?

Reply:
None means we cannot decide whether it is categorical or continuous.

Reply:
If we don't know if a column is categorical, just return False. Real numbers contains all discrete set of numbers (superset).